src/goto-programs/goto_program_template.h

```diff
@@ -516,99 +478,60 @@ void goto_program_templatet<codeT, guardT>::compute_loop_numbers()
 }
 
 template <class codeT, class guardT>
-void goto_program_templatet<codeT, guardT>::get_successors(
-  targett target,
-  targetst &successors)
+template <typename Target>
+std::list<Target> goto_program_templatet<codeT, guardT>::get_successors(
```

Comment:
This returns a list, which should either use move semantics (fast), or be constructed directly in the calling scope via RVO (really fast), so I doubt there's any penalty to this vs. modifying a list through a mutable reference.